### DISCUSSION TITLE
drag files into mindmap creates specific folder icon

### DISCUSSION THREAD

**Discussion by nnako** [freeplane/freeplane#1772] [Help] [answered]
Hi,

I often use Freeplane mindmaps to represent file structures. In order to createthese structures (at least on Windows computers), you can simply drag / drop the selected file(s) or folder(s) from a file explorer into the Freeplane mindmap. There, those files / folders will be represented showing the following attributes:

- the node's core shows the file's / folder's name
- the node's hyperlink is set to point to the respective file
- a "folder" icon is created in place of the regular "hyperlink" icon
- the regular (invisible) attributes are set: style_ref, id, created, modified

the lower part of the following image shows the result after a couple of drag / drop actions, some further styling and additional gray structure nodes:

![image](https://github.com/freeplane/freeplane/assets/5845602/79acdf04-e57e-4f99-a601-642554ef1dfb)

Now, when I don't drag / drop the files / folders into my map, I could usually achieve the same results by manually creating a new node, setting its textual content (i.e. core or text attribute) and then add the file's / folder's path into the node's link portion. On Windows, opening the link portion for this purpose is easily done using `<CTRL> + <K>` .

Although, I would expect the "folder" icons to appear for each node, I just see the regular external hyperlink icon (as seen in the upper part of the image). Even within the mindmap's resulting XML file I can not spot the difference of these two views. At least not by looking at the respective nodes. I would not assume that there is further metadata stored within the XML files to determine the icons for external hyperlinks.

Can somebody explain the cause for this effect? Is there a way to transform an "folder" icon into a "regular external hyperlink" icon or vice versa? Using the icon editor (`<CTRL> + <F2>` does not seem to provide modifying the icons in this case).

Thanks.

**Comment by macmarrum** [comment, marked as answer]
I think Freeplane looks for a slash at the end of an URI and checks if the directory exists before showing the folder icon.

I can also see the following info in [File#toURI()]( https://docs.oracle.com/javase/8/docs/api/java/io/File.html#toURI--)

> If it can be determined that the file denoted by this abstract pathname is a directory, then the resulting URI will end with a slash.

Do your folder URIs (shown via `Ctrl+K`) end with a slash?

**Reply by macmarrum** [reply]
There's also the setting `Preferences…->Appearance->Icons->Show link specific icons`

**Comment by nnako** [comment]
Hi @macmarrum ,

thanks. Yes, the (missing) `forward slash` at the end of the path was the detail I missed, so far.
